drivers/can/can_mcp251xfd.c

```diff
@@ -1394,6 +1394,12 @@ static inline int mcp251xfd_init_iocon_reg(const struct device *dev)
 		tmp |= MCP251XFD_REG_IOCON_SOF;
 	}
 
+	if (dev_cfg->xstby_enable) {
+		tmp &= ~MCP251XFD_REG_IOCON_TRIS0;
+		tmp &= ~MCP251XFD_REG_IOCON_PM0;
```

Comment:
It wasn't fully clear to me from the datasheet if the pin mode (PM0) has to be set to 0 or 1 for the XSTBY function, but if you have tested that it works with this setting, it should be correct.

![image](https://github.com/user-attachments/assets/b0348152-6a0a-4ee5-a750-cbffbeff6ecf)